### PR TITLE
feat: market module

### PR DESCRIPTION
**Pull request by Mettwasser**
### What did you fix? 
Adds a very simple implementation of the market API.
This will later be changed in v6.0.0 when market API v2 releases.

---

### Reproduction steps

---

### Considerations
- Does this contain a new dependency? **[Yes]**
- Does this introduce opinionated data formatting or manual data entry? **[No]**
- Does this pr include updated data files in a separate commit that can be reverted for a clean code-only pr? **[Yes]**
- Have I run the linter? **[Yes]**
- Is is a bug fix, feature request, or enhancement? **[Feature]**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "warframe"
-version = "5.0.0"
+version = "5.0.1"
```
Comment (TobiTenno) [inline Cargo.toml:3]: isn't this already in master? looks like you've got some excess commits. probably need to rebase

File: src/market/client.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        if response.status().is_success() {
+            let json_result = response.json::<ItemInfoPayload>().await?;
+            Ok(json_result.payload.item)
+        } else {
+            Err(response.status().into())
+        }
+    }
+
+    /// Fetches all tradable items
+    pub async fn items(&self) -> Result<Vec<Item>, ApiError> {
```
Comment (TobiTenno) [inline src/market/client.rs:61]: you'll probably wanna ratelimit this one yourself or catch ratelimit errors (HTTP 429), and cache this endpoint, cause it tends to be slow if you overuse it

**Review by TobiTenno — CHANGES_REQUESTED** [review]
very cool. just a few little things

File: src/market/client.rs (hunk 2 above)
Comment (Mettwasser) [reply src/market/client.rs:61]: I see. Yea, Ratelimiting is prolly a good idea. Alltho, I'd make the cache optional via a feature flag - maybe someone wants to implement their own?
Yeaaa, I kinda messed up the first rebase. Will do as soonnas I'm home

Comment (TobiTenno) [reply src/market/client.rs:61]: well, problem on the cache is i'm more getting at you probably wanna cache it so if you get a 429, you can return the last result and add some additional response info that you returned from cache due to ratelimit

Comment (TobiTenno) [reply src/market/client.rs:61]: i can rebase for you if you want

Comment (TobiTenno) [reply src/market/client.rs:61]: i did some rebase

**Review by TobiTenno — APPROVED** [review]

**Pull request merged**